### PR TITLE
add OLM install integration

### PR DESCRIPTION
**Pull request by ycao56**
1. added bundle files
1. added make target for generating bundle and index image
1. added instructions in readme
1. also added missing permission for `gatekeepers/finalizers`

### CODE REVIEW

**Comment by ycao56** [comment]
@font can you help to add github actions to build/push bundle and index images? I suggest we make it available on quay so that users can use it directly.

File: README.md [hunk 1]
```diff
@@ -55,3 +55,57 @@ registry service like [quay.io](https://quay.io).
...
+    make docker-push IMG=<index image name>
+    ```
+1. Create the CatalogSource/OperatorGroup/Subscription
+    ```yaml
+    ---
+    apiVersion: operators.coreos.com/v1alpha1
+    kind: CatalogSource
+    metadata:
+      name: gatekeeper-operator
+      namespace: openshift-marketplace
```
Comment (font) [inline README.md:88]: Should this be `gatekeeper-system`? If not, can we avoid using `openshift`? `operator-sdk` installs OLM in the `olm` namespace and creates an `operators` namespace, so perhaps we can use `operators`.

**Review by font — COMMENTED** [review]
Great to see this, thanks @ycao56! I made a few comments, but mostly just suggest running this against latest master.

File: README.md [hunk 2]
```diff
@@ -55,3 +55,57 @@ registry service like [quay.io](https://quay.io).
     ```shell
     make deploy IMG=<registry>/<imagename>:<tag>
     ```
+
+### Deploy Operator using OLM
+
+If you would like to deploy Operator using OLM, you'll need to build and push the bundle image and index image. You need to host the images on a public registry service like [quay.io](https://quay.io).
```
[the remark this answers is not in the document]
Comment (ycao56) [reply README.md:61]: that's correct. once the images are published, we can update this for development purpose

File: bundle/manifests/operator.gatekeeper.sh_gatekeepers.yaml [hunk 3]
```diff
@@ -0,0 +1,586 @@
+apiVersion: apiextensions.k8s.io/v1
```
[the remark this answers is not in the document]
Comment (ycao56) [reply bundle/manifests/operator.gatekeeper.sh_gatekeepers.yaml:1]: done

File: README.md (hunk 1 above)
Comment (ycao56) [reply README.md]: I changed namespace to gatekeeper-system for catalogsource and also update it in operatorgroup

File: README.md [hunk 4]
```diff
@@ -55,3 +55,57 @@ registry service like [quay.io](https://quay.io).
...
+    ```shell
+    make bundle-build REPO=<registry>
+    ```
+1. Push the bundle image
+    ```shell
+    make docker-push IMG=<bundle image name>
+    ```
+1. Build the index image
+
+    OPM needs to be installed first. Get it from [here](https://github.com/operator-framework/operator-registry/releases).
```
[the remark this answers is not in the document]
Comment (ycao56) [reply README.md:73]: yeah, I can add it in a separate PR

File: Makefile [hunk 5]
```diff
@@ -3,8 +3,12 @@ SHELL := /bin/bash
 VERSION ?= 0.0.1
 # Current Gatekeeper version
 GATEKEEPER_VERSION ?= v3.1.1
+# Default image repo
+REPO ?= quay.io/gatekeeper
 # Default bundle image tag
-BUNDLE_IMG ?= controller-bundle:$(VERSION)
+BUNDLE_IMG ?= $(REPO)/gatekeeper-operator-bundle:$(VERSION)
```
[the remark this answers is not in the document]
Comment (ycao56) [reply Makefile:9]: I will leave it as is then

**Review by font — COMMENTED** [review]
LGTM, thanks @ycao56! Can you squash the merge commit using a rebase instead?

**Review by font — APPROVED** [review]

**Pull request merged**